prog/vm/nexus.rb

```diff
@@ -295,6 +295,7 @@ def before_run
     out = `ssh -o BatchMode=yes -o ConnectTimeout=1 -o PreferredAuthentications=none user@#{addr} 2>&1`
     if out.include? "Host key verification failed."
       vm.update(display_state: "running")
+      Clog.emit("vm provisioned") { {vm: vm.values, provision: {vm_ubid: vm.ubid, vm_host_ubid: host.ubid, duration: Time.now - vm.created_at}} }
```

Comment:
Seems like between `thread` and `vm.values` we don't need `vm_ubid` either. ... Do we even need vm_host_ubid? Isn't it in `vm.values`? I guess not in ubid form.

Seems like I should add special handling in Clog to deal with Sequel models, e.g. automatically ubidifying uuids.

Reply:
Yeah, they are uuid not ubid. But it's not very important. I can refactored as `{vm: vm.values, provision: {duration: Time.now - vm.created_at}`

Reply:
maybe we can leave it out until we get annoyed chasing uuid->ubid conversions too often. I wonder if I should automatically rewrite all uuids as ubids, as a clog feature. to be continued.

Reply:
I leave it. I think "uuid->ubid" conversion as clog feature sounds good. We may add a clog representation into the models. Although `﻿model.values` returns numerous properties, a lightweight version would suffice for our needs